app/assets/javascripts/notebook/magic/customC3Chart.coffee

```diff
@@ -32,17 +21,15 @@ define([
     console.log("chartOptions", chartOptions)
     chartOptions.bindto = "#" + chart_container.attr("id")
     chartOptions.data = {} if not chartOptions.data
-    chartOptions.data.rows = []
+    chartOptions.data.json = @dataInit
 
-    chart = c3.generate(chartOptions)
+    chartOptions.data.keys = { value: options.headers } if not chartOptions.data.keys or not chartOptions.data.keys.value
 
-    chart.load({ rows: prepare_c3_data(options.headers, @dataInit) })
-    chart.flush()
+    chart = c3.generate(chartOptions)
 
     dataO.subscribe( (newData) =>
-      chart.unload();
-      chart.load({
-        rows: prepare_c3_data(options.headers, newData)
-      });
+      chartOptions.data.json = newData#_.filter(newData, (obj) -> !_.findWhere(currentData, obj))
+      #chartOptions.data.unload = true
```

Comment:
I think the data duplicates are managed by the `load`, I did a few tests and it worked but I'll do some more, but if you have one in head specific that you'd like to try let me know.

Changing the limit using the input text, that's what you said right? I'll give this a try first